[binarycaching] Pushing NuGet to GitHub failed

**Describe the bug**
Linux and macOS builds too long, but Nuget caching is on. Reason: Pushing NuGet to GitHub failed
```
2021-12-04T11:58:28.8727300Z Uploading binaries for boost-log:x64-osx-lgpl to NuGet source GitHub.
2021-12-04T11:58:33.0621450Z Pushing NuGet to GitHub failed. Use --debug for more information.
2021-12-04T11:58:33.0721380Z Installing package boost-log[core]:x64-osx-lgpl...
2021-12-04T11:58:33.0739250Z Elapsed time for package boost-log:x64-osx-lgpl: 2.087 min
```

**Environment**
- CI: Github Actions
- OS: linux, macos
- Compiler: Apple clang version 12.0.0 (clang-1200.0.32.29)

**To Reproduce**
Steps to reproduce the behavior:

0. Set global environment `VCPKG_BINARY_SOURCES="clear;nuget,GitHub,readwrite"`
0. Bootstrap ```thirdparty/vcpkg/bootstrap-vcpkg.sh -disableMetrics```
0. Store credentials
   ```
   mono `thirdparty/vcpkg/vcpkg fetch nuget | tail -n 1` sources add -source "https://nuget.pkg.github.com/wowcube/index.json" -storepasswordincleartext -name "GitHub" -username "wowcube" ***
   ```
0. CMake configure


**Expected behavior**
Packages are pushed and pulled to and from NuGet, build is fast as Windows.

**Failure logs** 
```
2021-12-04T11:55:14.6365830Z ##[group]Run set -x
2021-12-04T11:55:14.6366450Z [36;1mset -x[0m
2021-12-04T11:55:14.6367090Z [36;1mthirdparty/vcpkg/bootstrap-vcpkg.sh -disableMetrics[0m
2021-12-04T11:55:14.6545520Z shell: /bin/bash --noprofile --norc -e -o pipefail {0}
2021-12-04T11:55:14.6546070Z env:
2021-12-04T11:55:14.6546650Z   VCPKG_BINARY_SOURCES: clear;nuget,GitHub,readwrite
2021-12-04T11:55:14.6547200Z ##[endgroup]
2021-12-04T11:55:14.7395020Z + thirdparty/vcpkg/bootstrap-vcpkg.sh -disableMetrics
2021-12-04T11:55:14.8524750Z Downloading vcpkg-macos...
2021-12-04T11:55:15.6182260Z ##[group]Run mono `thirdparty/vcpkg/vcpkg fetch nuget | tail -n 1` sources add -source "https://nuget.pkg.github.com/wowcube/index.json" -storepasswordincleartext -name "GitHub" -username "wowcube" ***
2021-12-04T11:55:15.6185020Z [36;1mmono `thirdparty/vcpkg/vcpkg fetch nuget | tail -n 1` sources add -source "https://nuget.pkg.github.com/wowcube/index.json" -storepasswordincleartext -name "GitHub" -username "wowcube" ***[0m
2021-12-04T11:55:15.6403630Z shell: /bin/bash --noprofile --norc -e -o pipefail {0}
2021-12-04T11:55:15.6404260Z env:
2021-12-04T11:55:15.6404840Z   VCPKG_BINARY_SOURCES: clear;nuget,GitHub,readwrite
2021-12-04T11:55:15.6405440Z ##[endgroup]
2021-12-04T11:55:18.8123830Z Package source with Name: GitHub added successfully.
...
2021-12-04T11:55:18.9660990Z + cmake -B /Users/runner/work/FirmwareEmulator/FirmwareEmulator/cmake-build-release -DCMAKE_BUILD_TYPE=release -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/Users/runner/work/FirmwareEmulator/FirmwareEmulator/cmake-install-release -DCMAKE_TOOLCHAIN_FILE=thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_OVERLAY_TRIPLETS=cmake/triplets -DVCPKG_TARGET_TRIPLET=x64-osx-lgpl -DCMAKE_C_COMPILER:FILEPATH=clang -DCMAKE_CXX_COMPILER:FILEPATH=clang++
2021-12-04T11:55:18.9778730Z -- Running vcpkg install
2021-12-04T11:55:21.3706250Z Detecting compiler hash for triplet x64-osx...
2021-12-04T11:55:25.6614830Z Detecting compiler hash for triplet x64-osx-lgpl...
2021-12-04T11:55:28.2595900Z The following packages will be built and installed:
2021-12-04T11:55:28.2699540Z   * boost-algorithm[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-algorithm/338f795a1e6d73ab25444f64fe7ba1d617128813
2021-12-04T11:55:28.2799910Z   * boost-align[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-align/5e733503d18c7e82478961e1155a40a95af0ab42
2021-12-04T11:55:28.2903440Z   * boost-any[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-any/cf8bdfd2ce434a981620383dcc07d46c0a071ca1
2021-12-04T11:55:28.3003500Z   * boost-array[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-array/fb008cce0de1c6a34ac5222d03d7e1069d135dc4
2021-12-04T11:55:28.3106840Z   * boost-asio[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-asio/9c5c591c857588b69354bf075019bc78ce4237bd
2021-12-04T11:55:28.3207610Z   * boost-assert[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-assert/db6b88ce332c6752f87f8f0c098a832e23610303
2021-12-04T11:55:28.3311020Z   * boost-atomic[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-atomic/c5f02ea8b6555b20b6cc8901bdae9e9628086640
2021-12-04T11:55:28.3412450Z   * boost-bind[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-bind/a1764edb5aaa4cf532e6e2d60d56420240dec410
2021-12-04T11:55:28.3515200Z   * boost-build[core]:x64-osx -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-build/d12d4a6f44a7356c4262b27adcda834fe893b576
2021-12-04T11:55:28.3616730Z   * boost-chrono[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-chrono/f0143b4dd81e3de932ea886e69efd196452d81b3
2021-12-04T11:55:28.3719880Z   * boost-concept-check[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-concept-check/9bdfb3a9b14b35dc1833bc559226c7925096ff57
2021-12-04T11:55:28.3819880Z   * boost-config[core]:x64-osx-lgpl -> 1.77.0#2 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-config/0f7b70c02bfb43bbdc8f62bd49c8a5339a65f48c
2021-12-04T11:55:28.3920030Z   * boost-container[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-container/9af11f40e9b9fc2ebdda7e8885f9f49dea1923d7
2021-12-04T11:55:28.4021700Z   * boost-container-hash[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-container-hash/76c8d3c46db10b6774d6986822e53d1140934667
2021-12-04T11:55:28.4124360Z   * boost-context[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-context/ec8ceb582e4ef9936ceb840c70b90dfb20703f1d
2021-12-04T11:55:28.4224220Z   * boost-conversion[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-conversion/5b591e95244353857eaade53621ee895b1d9e0a6
2021-12-04T11:55:28.4324020Z   * boost-core[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-core/a56c8546828ee16a442a336c2385f9986f8c055f
2021-12-04T11:55:28.4411290Z   * boost-coroutine[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-coroutine/a5afb95d36e4f0baa263193b00e9a1abc1f01454
2021-12-04T11:55:28.4414310Z   * boost-date-time[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-date-time/e2ca710f2484af3cbeee428e063a7e4269510110
2021-12-04T11:55:28.4455140Z   * boost-detail[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-detail/57dabb22a4f9dfe872c6f007c9dbae7bf0847514
2021-12-04T11:55:28.4476560Z   * boost-dynamic-bitset[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-dynamic-bitset/e5b7742c115d4e4a450062e30b1542c6fe4aac5d
2021-12-04T11:55:28.4580020Z   * boost-endian[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-endian/6950c45ffb4e112920b6dd50c1cddde07b30d61d
2021-12-04T11:55:28.4601830Z   * boost-exception[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-exception/d60d712de012657effc2314a5d8781e453097129
2021-12-04T11:55:28.4624400Z     boost-filesystem[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-filesystem/7665f5ed27513a334df5b6051cb72d076ce626c2
2021-12-04T11:55:28.4638340Z   * boost-foreach[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-foreach/95983cd3eb11558e24d1f63a68a2ee944da927df
2021-12-04T11:55:28.4657430Z   * boost-format[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-format/15c6449aa221b65139b38cc6c6ef91f32e2d2b1d
2021-12-04T11:55:28.4678550Z   * boost-function[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-function/092af2f7e20c516fc02e8a39d8887f2301bb1fd9
2021-12-04T11:55:28.4681510Z   * boost-function-types[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-function-types/7675770af935fb375c0350c7a5fc561f9bc31432
2021-12-04T11:55:28.4686160Z   * boost-fusion[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-fusion/e33ce42cfab1b757d73fc0c4ac7edacaefd0be58
2021-12-04T11:55:28.4693450Z   * boost-integer[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-integer/375a2af7e2d5d620aa9f42064255c269b0c73f5a
2021-12-04T11:55:28.4722500Z   * boost-interprocess[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-interprocess/6461c4f63a5c0f31efb213c9d1e3b9328b2e3dcc
2021-12-04T11:55:28.4751090Z   * boost-intrusive[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-intrusive/9fb49266dc300d8914a0485972738b960ccf4339
2021-12-04T11:55:28.4760500Z   * boost-io[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-io/83ec8bab1f1bbd6eb1e2883a26747c7982d5c34f
2021-12-04T11:55:28.4774830Z   * boost-iterator[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-iterator/7bec2abc5b714c2bd975178406937ef59a73f29b
2021-12-04T11:55:28.4778700Z   * boost-lexical-cast[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-lexical-cast/3a265d9296755adc81f3c02b273d758798903b0c
2021-12-04T11:55:28.4782980Z     boost-log[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-log/6ab1ab88f37500f86b5c9b9222e50dc2c17da5c1
2021-12-04T11:55:28.4786460Z   * boost-modular-build-helper[core]:x64-osx -> 1.77.0#4 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-modular-build-helper/aea8b4dbb8063db29d8ac843ef6aac35478bebaa
2021-12-04T11:55:28.4790640Z   * boost-move[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-move/d912a2a9e0b705919dae22cd8af6a4467e209e4e
2021-12-04T11:55:28.4793730Z   * boost-mp11[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-mp11/f685c323f00cf309294ac9e26b554baacbe3aecf
2021-12-04T11:55:28.4797180Z   * boost-mpl[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-mpl/1b0bd80c486b32952b76c2b9539555cff5d2f911
2021-12-04T11:55:28.4800500Z   * boost-multi-index[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-multi-index/e993a57f4569f76310bc8c203372ebe472395005
2021-12-04T11:55:28.4805040Z   * boost-numeric-conversion[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-numeric-conversion/78140f78c6400520bd4ce340c68236e083a504e9
2021-12-04T11:55:28.4808230Z   * boost-optional[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-optional/4a1ddf516bc0edc855ab4d9b8d134f338ba51a94
2021-12-04T11:55:28.4811590Z   * boost-parameter[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-parameter/3f480980e4e1ce4713a28bd24d728efb0bb931fa
2021-12-04T11:55:28.4815200Z   * boost-phoenix[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-phoenix/2926d1b537106cb231bb9f3a657f6fde6c8530f8
2021-12-04T11:55:28.4818020Z   * boost-pool[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-pool/7794803f75109db0ca64a48d2f1cbd76100b153d
2021-12-04T11:55:28.4822210Z   * boost-predef[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-predef/a291dbfb68391a25f04944f9caa924f570b26d41
2021-12-04T11:55:28.4825400Z   * boost-preprocessor[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-preprocessor/6abad213e30df5ed168164b2612dba2041c7e8c1
2021-12-04T11:55:28.4828460Z     boost-process[core]:x64-osx-lgpl -> 1.77.0#2 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-process/0e476269a7091861c1db7379d4e925df6d184044
2021-12-04T11:55:28.4832990Z   * boost-property-tree[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-property-tree/959191d822cc095e02a3f6d9af5b68aeb823a0d1
2021-12-04T11:55:28.4836340Z   * boost-proto[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-proto/5d33f62f1117da9bac418c4b85d1a1bd96c547a4
2021-12-04T11:55:28.4841080Z   * boost-random[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-random/3f6c3399689c43f599f3852d98325cb272e8f775
2021-12-04T11:55:28.4843600Z   * boost-range[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-range/737f4f322e661e01459862fbe73f748390b59c3d
2021-12-04T11:55:28.4846920Z   * boost-ratio[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-ratio/724fa7b1c0e1d78a9c29595b042443090e53a8ee
2021-12-04T11:55:28.4850140Z   * boost-rational[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-rational/d54d04de158ddaf33d7b8a6815532e8ee2771515
2021-12-04T11:55:28.4854680Z     boost-regex[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-regex/c7c4497156a2c9e4f3227ee3807512630242307d
2021-12-04T11:55:28.4858500Z   * boost-serialization[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-serialization/bdb25d2d55569534825b139f62cf29a2f7aaba6e
2021-12-04T11:55:28.4861690Z   * boost-smart-ptr[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-smart-ptr/700532550b70d03254c77f1467ab4c01a2c6e62a
2021-12-04T11:55:28.4866090Z   * boost-spirit[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-spirit/bc66c5a9ccd7f7bde95426392bc7ff62189f920d
2021-12-04T11:55:28.4869140Z     boost-stacktrace[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-stacktrace/8d5ffa9ca1552ad32a5c6a78c28a3e275323dc2d
2021-12-04T11:55:28.4872520Z   * boost-static-assert[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-static-assert/57ca16ab3652380b32a137add7aed6a385754d75
2021-12-04T11:55:28.4877350Z   * boost-system[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-system/86ac553606d3af5cf67b96bafd1ddaf9e6c57b60
2021-12-04T11:55:28.4880140Z   * boost-thread[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-thread/1c097a032a997463e15245272e40ea9298554f8b
2021-12-04T11:55:28.4883220Z   * boost-throw-exception[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-throw-exception/f0887800c8710d96015c7c1541f78265e1227963
2021-12-04T11:55:28.4887660Z   * boost-tokenizer[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-tokenizer/56a1fb828a469ba50000983860226452c264050f
2021-12-04T11:55:28.4890520Z   * boost-tuple[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-tuple/35d10ab4f156848f6f08f8bc26e41b27638af55e
2021-12-04T11:55:28.4893630Z   * boost-type-index[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-type-index/b8c5d86d75e307f04c4e5548f2ff57dccdc29e40
2021-12-04T11:55:28.4896780Z   * boost-type-traits[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-type-traits/7f108bb5313d004627d2b81b5e2a6d7298b091e9
2021-12-04T11:55:28.4901360Z   * boost-typeof[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-typeof/7df16df37b9fe9d5e4184dfdf928030010522f51
2021-12-04T11:55:28.4905280Z   * boost-uninstall[core]:x64-osx -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-uninstall/1d472d1b44489c230a741cb519e893732f5d00ac
2021-12-04T11:55:28.4908360Z   * boost-uninstall[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-uninstall/1d472d1b44489c230a741cb519e893732f5d00ac
2021-12-04T11:55:28.4911620Z   * boost-unordered[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-unordered/e0e056b82651533817c1ca6dce6fb6239ccdcdbb
2021-12-04T11:55:28.4914790Z   * boost-utility[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-utility/e3589215d60cac29105fb361ef70e6d8befb8f94
2021-12-04T11:55:28.4919020Z   * boost-variant[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-variant/63d66b780f8fb2e1b7c1365fc54b23795b09279c
2021-12-04T11:55:28.4922180Z   * boost-vcpkg-helpers[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-vcpkg-helpers/f0b625fd73edd33c92f70727a4cf068d0268379e
2021-12-04T11:55:28.4925340Z   * boost-winapi[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-winapi/2d4c0be143dbffd619ac154ef5d8f4fe24277bbe
2021-12-04T11:55:28.4928360Z   * boost-xpressive[core]:x64-osx-lgpl -> 1.77.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-xpressive/bf04b7076a27d24203528b52aaf41efc83923c67
2021-12-04T11:55:28.4933220Z     cppzmq[core]:x64-osx-lgpl -> 4.8.1 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/cppzmq/79921528ad838f11245ebfb6b3f2fe9dc4bdca26
2021-12-04T11:55:28.4936450Z   * openssl[core]:x64-osx-lgpl -> 1.1.1l#3 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/openssl/83369bdefab234d8570ac2c0f3b616b0c4ab6cdf
2021-12-04T11:55:28.4940860Z     protobuf[core]:x64-osx-lgpl -> 3.18.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/protobuf/19c2bf45c235e6126161bae36aa7ff7e1fcda479
2021-12-04T11:55:28.4943430Z   * protobuf[core]:x64-osx -> 3.18.0 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/protobuf/19c2bf45c235e6126161bae36aa7ff7e1fcda479
2021-12-04T11:55:28.4946000Z   * vcpkg-cmake[core]:x64-osx -> 2021-09-13 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/vcpkg-cmake/fc4d9fcc5b8d2b97c083c6b70dd06df5174bd97b
2021-12-04T11:55:28.4950120Z   * vcpkg-cmake-config[core]:x64-osx -> 2021-11-01 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/vcpkg-cmake-config/38a87ee8edd9ea8e8fff604fbcb785661a8d0e28
2021-12-04T11:55:28.4953390Z     zeromq[core]:x64-osx-lgpl -> 4.3.4#2 -- /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/zeromq/423c351e8759303bc9ba74a065ce9b983a5f85f5
2021-12-04T11:55:28.4955320Z Additional packages (*) will be modified to complete this operation.
2021-12-04T11:55:28.4956860Z Attempting to fetch 84 packages from nuget.
2021-12-04T11:56:27.7040840Z Restored 81 packages from NuGet in 53.02 s. Use --debug for more information.
2021-12-04T11:56:27.7141200Z Starting package 1/84: vcpkg-cmake:x64-osx
2021-12-04T11:56:27.7249510Z Installing package vcpkg-cmake[core]:x64-osx...
2021-12-04T11:56:27.7318480Z Elapsed time for package vcpkg-cmake:x64-osx: 5.514 ms
2021-12-04T11:56:27.7531400Z Starting package 2/84: boost-uninstall:x64-osx-lgpl
2021-12-04T11:56:27.7705480Z Installing package boost-uninstall[core]:x64-osx-lgpl...
2021-12-04T11:56:27.7809030Z Elapsed time for package boost-uninstall:x64-osx-lgpl: 2.523 ms
2021-12-04T11:56:27.7908740Z Starting package 3/84: boost-vcpkg-helpers:x64-osx-lgpl
2021-12-04T11:56:27.8012730Z Installing package boost-vcpkg-helpers[core]:x64-osx-lgpl...
...
2021-12-04T11:56:28.2495460Z Starting package 77/84: boost-log:x64-osx-lgpl
2021-12-04T11:56:28.2527650Z Building package boost-log[core]:x64-osx-lgpl...
2021-12-04T11:56:28.2584090Z -- [OVERLAY] Loading triplet configuration from: /Users/runner/work/FirmwareEmulator/FirmwareEmulator/cmake/triplets/x64-osx-lgpl.cmake
2021-12-04T11:56:28.2646130Z -- Installing port from location: /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/versioning_/versions/boost-log/6ab1ab88f37500f86b5c9b9222e50dc2c17da5c1
2021-12-04T11:58:25.0719000Z -- Downloading https://github.com/boostorg/log/archive/boost-1.77.0.tar.gz -> boostorg-log-boost-1.77.0.tar.gz...
2021-12-04T11:58:25.0721080Z -- Extracting source /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/downloads/boostorg-log-boost-1.77.0.tar.gz
2021-12-04T11:58:25.0722990Z -- Using source at /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/buildtrees/boost-log/src/ost-1.77.0-b54115eac1.clean
2021-12-04T11:58:25.0724220Z -- Found external ninja('1.10.2').
2021-12-04T11:58:25.0725410Z -- Configuring x64-osx-lgpl-dbg
2021-12-04T11:58:25.0726260Z -- Configuring x64-osx-lgpl-rel
2021-12-04T11:58:25.0727080Z -- Building x64-osx-lgpl-dbg
2021-12-04T11:58:25.0727870Z -- Building x64-osx-lgpl-rel
2021-12-04T11:58:25.0728630Z -- Packaging headers
2021-12-04T11:58:25.0729390Z -- Packaging headers done
2021-12-04T11:58:25.0731040Z -- Downloading https://raw.githubusercontent.com/boostorg/boost/boost-1.77.0/LICENSE_1_0.txt -> boost_LICENSE_1_0.txt...
2021-12-04T11:58:25.0733360Z -- Installing: /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/packages/boost-log_x64-osx-lgpl/share/boost-log/usage
2021-12-04T11:58:25.0735370Z -- Installing: /Users/runner/work/FirmwareEmulator/FirmwareEmulator/thirdparty/vcpkg/packages/boost-log_x64-osx-lgpl/share/boost-log/copyright
2021-12-04T11:58:25.0736700Z -- Performing post-build validation
2021-12-04T11:58:25.0737690Z -- Performing post-build validation done
2021-12-04T11:58:28.8727300Z Uploading binaries for boost-log:x64-osx-lgpl to NuGet source GitHub.
2021-12-04T11:58:33.0621450Z Pushing NuGet to GitHub failed. Use --debug for more information.
2021-12-04T11:58:33.0721380Z Installing package boost-log[core]:x64-osx-lgpl...
2021-12-04T11:58:33.0739250Z Elapsed time for package boost-log:x64-osx-lgpl: 2.087 min
...
```

**Additional context**
I see 240 private packages in GitHub for windows, linux and macos. Windows is not affected somehow.

```
2021-12-17T12:47:34.4214580Z [65/191] Building CXX object CMakeFiles/rtti_obfuscator.dir/tool/rtti_obfuscator.cc.o
2021-12-17T12:47:34.4316620Z FAILED: CMakeFiles/rtti_obfuscator.dir/tool/rtti_obfuscator.cc.o 
2021-12-17T12:47:34.4422210Z /usr/bin/clang++  -isystem /Users/runner/work/FirmwareEmulator/FirmwareEmulator/cmake-build-release/vcpkg_installed/x64-osx-lgpl/include -O3 -DNDEBUG -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.13 -std=gnu++2a -MD -MT CMakeFiles/rtti_obfuscator.dir/tool/rtti_obfuscator.cc.o -MF CMakeFiles/rtti_obfuscator.dir/tool/rtti_obfuscator.cc.o.d -o CMakeFiles/rtti_obfuscator.dir/tool/rtti_obfuscator.cc.o -c /Users/runner/work/FirmwareEmulator/FirmwareEmulator/tool/rtti_obfuscator.cc
2021-12-17T12:47:34.4528360Z /Users/runner/work/FirmwareEmulator/FirmwareEmulator/tool/rtti_obfuscator.cc:88:23: error: no matching constructor for initialization of 'std::ifstream' (aka 'basic_ifstream<char>')
2021-12-17T12:47:34.4630830Z         std::ifstream ifs(path, std::fstream::binary);
2021-12-17T12:47:34.4732570Z                       ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/docs/users/binarycaching.md
+++ b/docs/users/binarycaching.md
@@ -96,14 +96,17 @@ steps:
   - name: 'Setup NuGet Credentials'
     shell: 'bash'
     # Replace <OWNER> with your organization name
-    run: >
-      ${{ matrix.mono }} `./vcpkg/vcpkg fetch nuget | tail -n 1`
-      sources add
-      -source "https://nuget.pkg.github.com/<OWNER>/index.json"
-      -storepasswordincleartext
-      -name "GitHub"
-      -username "<OWNER>"
-      -password "${{ secrets.GITHUB_TOKEN }}"
+    run: |
+      ${{ matrix.mono }} `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+        sources add \
+        -source "https://nuget.pkg.github.com/<OWNER>/index.json" \
+        -storepasswordincleartext \
+        -name "GitHub" \
+        -username "<OWNER>" \
+        -password "${{ secrets.GITHUB_TOKEN }}"
+      ${{ matrix.mono }} `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+        setapikey "${{ secrets.GITHUB_TOKEN }}" \
+        -source "https://nuget.pkg.github.com/<OWNER>/index.json"
 
   # Omit this step if you're using manifests
   - name: 'vcpkg package restore'

